packages/ts/src/components/treemap/types.ts

```diff
@@ -0,0 +1,14 @@
+import { HierarchyRectangularNode } from 'd3-hierarchy'
+
+export type TreemapDatum<Datum> = {
+  datum: Datum;
+  index: number;
+}
+
+export interface TreemapNode<Datum> extends HierarchyRectangularNode<TreemapDatum<Datum>> {
+  _id: string;
+  _state?: {
```

Comment:
I know `NestedDonut` has a similar type, but do we really need this `_state` as a dedicated nested object?

I feel like we can flatten it and that will make the code look cleaner (by removing `_state?.` fragments).

Reply:
I agree. I was also skeptical about pulling in this logic from the nested donut. Will try to get rid of this.

Reply:
It's tricky to remove this, because it helps with propagating the colors from the parents to the children.

I tried computing the color at render time, but it's more complex since we'd need to traverse up the tree for each node to figure out what color it should be, based on its parents.

I say let's leave this logic here, as it's in a way cleaner to traverse the whole hierarchy just once to compute the colors in a dedicated step. Perhaps later we can clean it up in both the treemap and nested donut at the same time, if there's a cleaner way.